mne/io/eeglab/eeglab.py

```diff
@@ -226,6 +233,12 @@ class RawEEGLAB(_BaseRaw):
         Names or indices of channels that should be designated
         EOG channels. If 'auto', the channel names containing
         ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
+    event_id : dict
+        By default, events are read from the input file by dropping every
+        non-integer part of events containing integers, and completely
+        dropping any events without integer parts. If non-integer events
+        should be read, this should be a dict mapping from their names to
+        integers, e.g. dict(fmri_scan_onset=199, recording_start=255).
```

Comment:
Underlyingly, what you IMO want is something like

```
events = list()
for event_code in list_of_event_codes:
    parsed_code = event_id.get(event_code, event_id_func(event_code))
    events.append(parsed_code)
```

And for BV, event_id_func is *always* dropping leading alphabetic characters and returning the int part (with the exception of 3 or so events of the type `recording_break` or so). So for BV, `event_id_func` is unnecessary.

So I think having a separate `event_id` and `event_id_func` is clearer; first, look it up in an optional dict, and default to calling an optional function. For BV, the function is always the same, so it's not exposed. For EEGLAB, it defaults to the same one for BV, but it's exposed for dealing with other input types.

If you really think BV should have a `callable` option though, I'll ping Teon.